### PR TITLE
election: always try to clean last election key

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
close https://github.com/pingcap/dm/issues/1983

### What is changed and how it works?
when `Campaign` returned an error, always try to clean last election key

### Check List 

Tests 

 - hard to test

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note

### CODE REVIEW

File: tests/ha_master/run.sh [hunk 1]
```diff
@@ -58,14 +58,8 @@ function test_evict_leader() {
...
 		fi
-		# Leader evict is not effective immediately, we need to wait for a proper period of time.
-		for _ in {0..30}; do
-			NEW_LEADER_NAME=$(get_leader "$WORK_DIR" 127.0.0.1:${MASTER_PORT1})
-			if [ "$NEW_LEADER_NAME" != "$LEADER_NAME" ]; then
-				break
-			fi
-			sleep 1
-		done
+		NEW_LEADER_NAME=$(get_leader $WORK_DIR 127.0.0.1:${MASTER_PORT1})
```
Comment (lance6716) [inline tests/ha_master/run.sh:61]: revert #1946

**Review by glorv — APPROVED** [review]

File: tests/ha_master/run.sh (hunk 1 above)
Comment (gozssky) [reply tests/ha_master/run.sh:61]: https://github.com/pingcap/dm/pull/1946 shouldn't be reverted? What it focus on is different. The evict request is asynchronous, so new leader may not be elected immediately after the request returns.

File: pkg/election/election.go [hunk 2]
```diff
@@ -267,18 +267,14 @@ func (e *Election) campaignLoop(ctx context.Context, session *concurrency.Sessio
...
-				// sometimes we noticed auto Resign does not work, so we try to manually Resign.
-				e.l.Debug("before manually resign",
-					zap.String("current election key", elec.Key()),
-					zap.Any("current election header", elec.Header()))
-				resignCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
-				defer cancel()
-				if err3 := elec.Resign(resignCtx); err3 != nil {
-					e.l.Warn("failed to manually resign", zap.Error(err3))
+				// because inner commit may return undetermined error, we try to delete the election key manually
+				deleted, err3 := e.ClearSessionIfNeeded(ctx, e.ID())
```
Comment (Ehco1996) [inline pkg/election/election.go:271]: Can you please tell me why I need to delete this key manually ?

and in old logic, we do `Resign` if `Campaign` failed, after this change ,why don't we do this anymore?

Comment (lance6716) [reply pkg/election/election.go:271]: see the issue in PR description

**Review by Ehco1996 — APPROVED** [review]

**Pull request merged**